actions/setup-nix/action.yml

```diff
@@ -0,0 +1,134 @@
+
+inputs:
+  # custom cache inputs ----------------------------------
+  cache-url:
+    description: "Custom nix cache URL"
+    required: false
+    default: ""
+  cache-pubkey:
+    description: "Custom nix cache binary public key"
+    required: false
+    default: ""
+
+  # AWS inputs ------------------------------------
+  enable-aws:
+    description: "Enable AWS credentials for nix cache"
+    required: false
+    default: false
+  aws-region:
+    description: ""
+    required: false
+    default: "us-west-1"
+  role-to-assume: 
+    description: ""
+    required: false
+    default: ""
+  role-duration-seconds:
+    description: ""
+    required: false
+    default: 3600
+    
+  # cachix inputs --------------------------------
+  enable-cachix:
+    description: "Run cachix for nix caching"
+    required: false
+    default: false
+  cachix-name:
+    description: "Cachix private cache name"
+    required: false
+    default: "manifest"
+  cachix-token:
+      description: "[optional] Cachix auth token for private cache"
+      required: false
+      default: ""
+
+  # github inputs ---------------------------------
+  github-token:
+    description: "GitHub auth token for private repos (if needed)"
+    required: false
+    default: ""
+
+  # grafana inputs --------------------------
+  metrics-job-name:
+    description: "grafana metrics job name"
+    required: false
+    default: manifest-build
+  gc-host:
+    description: "grafana hostname"
+    required: false
+  gc-basic-auth:
+    description: "grafana basic auth"
+    required: false
+  gc-org-id:
+    description: "grafana org/tenant id"
+    required: false
+
+runs:
+  using: composite
+  steps:
+    - name: configure AWS credentials
+      if: inputs.enable-aws == 'true'
+      uses: aws-actions/configure-aws-credentials@010d0da01d0b5a38af31e9c3470dbfdabdecca3a # v4.0.1
+      with:
+        role-to-assume: ${{ inputs.role-to-assume }}
+        role-duration-seconds: ${{ inputs.role-duration-seconds }}
+        aws-region: ${{ inputs.aws-region }}
+    - name: setup AWS credentials for nix
+      if: inputs.enable-aws == 'true'
+      shell: bash
+      run: |
+        # set up folder with AWS configs
+        aws configure set aws_access_key_id $AWS_ACCESS_KEY_ID
+        aws configure set aws_secret_access_key $AWS_SECRET_ACCESS_KEY
+        aws configure set aws_session_token $AWS_SESSION_TOKEN
+        aws configure set region ${{ inputs.aws-region }}
+
+        # determine root path
+        ROOT_PATH=""
+        if [[ "${{ runner.os }}" == "Linux" ]]; then
+          ROOT_PATH="/root"
+        elif [[ "${{ runner.os }}" == "macOS" ]]; then
+          ROOT_PATH="/var/root"
+        else
+          echo "${{ runner.os }} not supported"
+          exit 1
+        fi
+
+        # symlink credential folders for nix build access
+        sudo ln -s ~/.aws $ROOT_PATH
+
+    - uses: cachix/install-nix-action@6004951b182f8860210c8d6f0d808ec5b1a33d28 # v25
```

Comment:
basically this provides a single CI action that can be used to setup nix + caches across any repo. it's already composable with mixing various caches together - so splitting it further seems like a hit to cleanliness?

this way, we can also enforce the same nix version across CIs rather than each CI specifying their own?